### PR TITLE
Updated Create new job form & added tests

### PR DESCRIPTION
**Pull request by benmenachem**
Updated the create new job form to be function based and not based on CreateViews
Also added tests for the function together with @guyy13

### CODE REVIEW

**Review by guyy13 — APPROVED** [review]
LGTM

File: jobboard/tests.py [hunk 1]
```diff
@@ -182,3 +186,57 @@ def test_page_display_to_user_that_is_recruiter(client):
...
+
+@pytest.mark.django_db
+def test_create_new_job_form_with_invalid_data(example_recruiter, example_job_data_as_dictionary,
+                                               client):
+    example_job_data_as_dictionary['title'] = ""  # invalid title
+    client.force_login(example_recruiter.user)
+    response = client.post("/recruiter/create_new_job_form/", data=example_job_data_as_dictionary)
+    assert response.status_code == 200
+    form = response.context["form"]
+    assert form.is_valid() is False
```
Comment (ifireball) [inline jobboard/tests.py]: Missing test for the "simple empty form display" code path.

Also there is no test that the form is actually displayed to the end user (e.g. make sure that the right template is used)

**Review by eldaraharonson — APPROVED** [review]

**Review by yaelerdr — APPROVED** [review]

**Review by YarinBou — APPROVED** [review]

File: SJMaster/urls.py [hunk 2]
```diff
@@ -16,26 +16,22 @@
...
 from jobboard.views import board
+from recruiter.views import create_new_job_form
 from student.views import update_student_account_settings_view, account_update_success
 from recruiter.views import (UpdateRecruiterSettings,
-                             CreateNewJobForm,
                              job_created_successfully,
                              recruiterRegister,
                              recruiter_created_successfully,
-                             recruiter_view_my_jobs_and_applications,)
+                             recruiter_view_my_jobs_and_applications, )
```
Comment (guy9050) [inline SJMaster/urls.py]: what's this space for?

File: jobboard/tests.py [hunk 3]
```diff
@@ -182,3 +186,57 @@ def test_page_display_to_user_that_is_recruiter(client):
...
+                   "work_from": 'Office-Only',
+                   "description": 'Junior software engineer',
+                   "city": jobboard.models.City.objects.get(name="Tel Aviv-Yafo").id,
+                   "address": 'Hashalom',
+                   "title_keywords": [jobboard.models.JobTitleKeyword.objects.get(keyword="software").id]}
+    return example_job
+
+
+@pytest.mark.django_db
+def test_create_new_job_form_with_valid_data(example_recruiter, example_job_data_as_dictionary,
```
Comment (guy9050) [inline jobboard/tests.py]: Change the parameters in the order that they are used, i.e `client` is used first so it should be the first one

File: jobboard/tests.py [hunk 4]
```diff
@@ -182,3 +186,57 @@ def test_page_display_to_user_that_is_recruiter(client):
...
+        "title_keywords"]
+
+
+@pytest.mark.django_db
+def test_create_new_job_form_with_invalid_data(example_recruiter, example_job_data_as_dictionary,
+                                               client):
+    example_job_data_as_dictionary['title'] = ""  # invalid title
+    client.force_login(example_recruiter.user)
+    response = client.post("/recruiter/create_new_job_form/", data=example_job_data_as_dictionary)
+    assert response.status_code == 200
```
Comment (guy9050) [inline jobboard/tests.py]: Do you get status code 200?

File: recruiter/views.py [hunk 5]
```diff
@@ -73,3 +56,19 @@ def recruiterRegister(request):
 
 def recruiter_created_successfully(request):
     return render(request, 'recruiter/recruiter_created_successfully.html')
+
+
+def create_new_job_form(request):
+    recruiter_object = get_object_or_404(Recruiter, user_id=request.user.id)
```
Comment (guy9050) [inline recruiter/views.py]: if it returns 404 your functions continues but then will fail/insert invalid data on lines 68-69

**Review by guy9050 — COMMENTED** [review]
I added some comments, otherwise looks good

File: SJMaster/urls.py (hunk 2 above)
Comment (benmenachem) [reply SJMaster/urls.py]: typo

File: jobboard/tests.py (hunk 3 above)
Comment (benmenachem) [reply jobboard/tests.py]: will do

File: jobboard/tests.py (hunk 4 above)
Comment (benmenachem) [reply jobboard/tests.py]: Yea I do get status code 200 since the data invalid so I'm staying at the same page. I believe that's what you spoke about with Barak yesterday in the lecture

File: recruiter/views.py (hunk 5 above)
Comment (benmenachem) [reply recruiter/views.py]: If I understand correctly of what I read online, if the user isn't found then get_object_or_404 will raise a 404 error and the function will stop and won't continue

**Review by ifireball — CHANGES_REQUESTED** [review]

File: recruiter/views.py (hunk 5 above)
Comment (ifireball) [reply recruiter/views.py]: Yes the 404 error is thrown as an exception so the function does not continue.

File: jobboard/tests.py (hunk 1 above)
Comment (benmenachem) [reply jobboard/tests.py]: @ifireball I updated the test to check that the form is displayed to the end user 
regarding test for "simple empty form display" I'm not sure I understand what you mean. do you mean you want me to test that the form is empty when the page is loaded?

File: recruiter/views.py [hunk 6]
```diff
@@ -74,4 +57,22 @@ def recruiterRegister(request):
 
 
 def recruiter_created_successfully(request):
+    jobboard.views.add_navbar_links_to_context(request,)
     return render(request, 'recruiter/recruiter_created_successfully.html')
+
+
+def create_new_job_form(request):
```
Comment (ifireball) [inline recruiter/views.py]: Why have this in `recruiter.views` and not in `jobbard.views`?

**Review by ifireball — CHANGES_REQUESTED** [review]

File: recruiter/views.py (hunk 6 above)
Comment (benmenachem) [reply recruiter/views.py]: @ifireball since it's a feature that only a recruiter has I thought it would make more sense to put it in the recruiter.views

Comment (ifireball) [reply recruiter/views.py]: The models, the forms that edit them, the views that show them and the tests for them typically all go in the same app. Where would you intuitively look for the view that creates new Job objects?

Comment (benmenachem) [reply recruiter/views.py]: I guess your right. to be honest, intuitively it made sense to me to put it in the recruiter views since it is a recruiter feature. But now I understand your point and it does make more sense to me to put it in the jobboard views. I will fix it now and update the PR.

Comment (benmenachem) [reply recruiter/views.py]: @ifireball fixed it and moved the function and the form to the jobboard app

File: jobboard/tests.py (hunk 1 above)
Comment (benmenachem) [reply jobboard/tests.py]: added the requested test

File: recruiter/views.py [hunk 7]
```diff
@@ -71,4 +48,5 @@ def recruiterRegister(request):
 
 
 def recruiter_created_successfully(request):
+    jobboard.views.add_navbar_links_to_context(request, )
```
Comment (guyy13) [inline recruiter/views.py]: This does nothing as the function needs to get a dictionary as well.
Anyway, I already have a PR open about this so you can remove this line.

Comment (benmenachem) [reply recruiter/views.py]: fixed it

**Review by guyy13 — APPROVED** [review]

File: jobboard/tests/model_function_tests.py [hunk 8]
```diff
@@ -112,3 +116,89 @@ def test_get_jobs_by_major():
...
+def test_navbar_links_are_correct_for_student_user(client, example_student_a):
+    client.force_login(example_student_a.user)
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.context['navbar_links'] == {"Account Settings": "/student/account_settings/", "Logout": "/logout",
+                                                f"Welcome {example_student_a.user.username}": "#"}
+
+
+@pytest.mark.django_db
+def test_navbar_links_are_correct_for_recruiter_user(client):
```
Comment (guyy13) [inline jobboard/tests/model_function_tests.py]: There's a separate file fo the nav_bar tests, not sure why you're adding all of these tests again.
I don't think you need to add anything to the model_function_tests file, only to the view_tests

File: jobboard/tests/conftest.py [hunk 9]
```diff
@@ -21,3 +24,27 @@ def example_student_a(example_institution_a):
                       major=Major.COMPUTER_SCIENCE, about="testing", graduation_date=date(2020, 7, 13))
     student.save()
     return student
+
+
+@pytest.fixture
+def example_recruiter():
```
Comment (guyy13) [inline jobboard/tests/conftest.py:30]: @ifireball said to keep the conftests file small, so maybe if these fixtures are only being used in the view_tests.py file it's better to put them in the file itself and not here

**Review by guyy13 — CHANGES_REQUESTED** [review]
left some comments, besides that lgtm

File: jobboard/tests/conftest.py (hunk 9 above)
Comment (benmenachem) [reply jobboard/tests/conftest.py:30]: I saw his comment on Eldar's PR but I think it makes more sense to put the fixtures here since it's just 2 more and we're probably not gonna put more fixtures in that file.

File: jobboard/tests/model_function_tests.py (hunk 8 above)
Comment (benmenachem) [reply jobboard/tests/model_function_tests.py]: fixed it.

File: jobboard/tests/model_function_tests.py [hunk 10]
```diff
@@ -3,6 +3,7 @@
 from student.models import Major
 from datetime import date
 
+
```
Comment (guyy13) [inline jobboard/tests/model_function_tests.py]: Can you remove this? No need to change this file if you don't add anything to it, it might cause unnecessary conflicts with different PRs

**Review by guyy13 — APPROVED** [review]

**Review by guy9050 — APPROVED** [review]
lgtm
